libzfs.pyx

```diff
@@ -2411,23 +2433,41 @@ cdef class ZFSObject(object):
             d.refresh()
             return d
 
-    def rename(self, new_name, nounmount=False, forceunmount=False, recursive=False):
-        cdef libzfs.renameflags_t flags
-        cdef const char *c_new_name = new_name
+    IF HAVE_RENAMEFLAGS_T:
 
-        flags.recurse = recursive
-        flags.nounmount = nounmount
-        flags.forceunmount = forceunmount
+        def rename(self, new_name, nounmount=False, forceunmount=False, recursive=False):
+            cdef libzfs.renameflags_t flags
+            cdef const char *c_new_name = new_name
 
-        cdef int ret
+            flags.recurse = recursive
+            flags.nounmount = nounmount
+            flags.forceunmount = forceunmount
 
-        with nogil:
-            ret = libzfs.zfs_rename(self.handle, NULL, c_new_name, flags)
+            cdef int ret
 
-        if ret != 0:
-            raise self.root.get_error()
+            with nogil:
+                ret = libzfs.zfs_rename(self.handle, NULL, c_new_name, flags)
+
+            if ret != 0:
+                raise self.root.get_error()
 
-        self.root.write_history('zfs rename', '-f' if forceunmount else '', '-u' if nounmount else '', self.name)
+            self.root.write_history('zfs rename', '-f' if forceunmount else '', '-u' if nounmount else '', self.name)
+
+    ELSE:
+
+        def rename(self, new_name, force_unmount=False, recursive=False):
```

Comment:
force_unmount is different from the other function define.

How about we use the same method scope but raise a RuntimeError if `nounmount` is used without HAVE_RENAMEFLAGS_T ?

Reply:
@william-gr i did that initially but then reverted to this. My reasoning at the time was that the function signature would show that we still accept nounmount ( even if we raise an exception ) - so i reverted to this case. Do you think that is okay and i can move back to having the same signature and raising an exception ?

Reply:
Yes, I think that makes things simpler in the long run.

Reply:
Perfect, updating asap